### PR TITLE
Fixes ifelse after is.na eval change

### PR DESCRIPTION
**Pull request by edgararuiz**
Changes exclusive to MS SQL

- Adds custom a custom `mssql_is_na()` function that equates `cond` to TRUE so that CASE/WHEN works properly.  Because the options are binary, the False/No test is simplified to use ELSE
- Add custom to `if`, `ifelse` and `if_else` 
- Add tests 

Used same tests as https://github.com/tidyverse/dbplyr/pull/51 to confirm nothing originally fixed broke, and also tested these sucesfully:

```
db_mtcars %>%
  group_by(x = ifelse(mpg < 20, "good", "bad")) %>%
  select(mpg, x)

db_mtcars %>%
  group_by(x = if_else(mpg < 20, "good", "bad")) %>%
  select(mpg, x)

db_mtcars %>%
  mutate(x = ifelse(mpg <= 20 & am == 0, "good", "bad"))
```

### CODE REVIEW

**Comment by edgararuiz** [comment]
PTAL @hadley

File: R/db-odbc-mssql.R [hunk 1]
```diff
@@ -218,4 +222,17 @@ mssql_generic_infix <- function(if_select, if_filter) {
   }
 }
 
+mssql_sql_if <- function(cond, if_true, if_false = NULL) {
+  build_sql(
+    "CASE WHEN ((", cond, ") =  'TRUE')", " THEN (", if_true, ")",
```
Comment (hadley) [inline R/db-odbc-mssql.R]: Can you please check to see if this has the same problem as fixed in tidyverse/dplyr#3157

Also can you have a go at using `sql_expr()` here? It should make the code quite a bit simpler.

**Comment by edgararuiz** [comment]
PTAL @hadley

**Comment by hadley** [comment]
Thanks!

**Pull request merged**